vnext/Microsoft.ReactNative/Views/TouchEventHandler.cpp

```diff
@@ -421,13 +421,33 @@ bool TouchEventHandler::TagFromOriginalSource(
       // while the content of the <Text> becomes a list of XAML <Run> elements.
       // However, we should report the Text element as the target, not the contexts of the text.
       if (const auto textBlock = sourceElement.try_as<xaml::Controls::TextBlock>()) {
+        if (textBlock.Inlines().Size() == 0) {
+          // No need to hit test if TextBlock does not use Inlines
+          break;
+        }
+
         const auto pointerPos = args.GetCurrentPoint(textBlock).RawPosition();
-        const auto inlines = textBlock.Inlines().GetView();
+        const auto textPointer = GetPositionFromPoint(textBlock, pointerPos);
+        if (!textPointer) {
+          // Did not find a matching character in the hit test
+          break;
+        }
 
-        bool isHit = false;
-        const auto finerTag = TestHit(inlines, pointerPos, isHit);
-        if (finerTag) {
-          tag = finerTag;
+        const auto childTag = textPointer.Parent().ReadLocalValue(xaml::FrameworkElement::TagProperty());
+        if (childTag != xaml::DependencyProperty::UnsetValue()) {
+          auto finerTag = childTag.try_as<winrt::IPropertyValue>();
+          if (finerTag) {
+            // React Native doesn't like when the target is a raw text node
```

Comment:
Maybe add a more descriptive comment:
"press handlers can be registered on top-level and nested Text elements, which are realized with the XAML TextBlock and Span elements, respectively"
Or something along these lines alluding to the comment  at 422.

Reply:
The issue isn't actually with the press handler, it's an unrelated issue in ScrollView.js. Raw text nodes in the React JS tree are always referred to by their tag and don't resolve to a React component.